plenum/test/consensus/helper.py

```diff
@@ -141,14 +142,24 @@ def __init__(self, node_count: int = 4, random: Optional[SimRandom] = None):
             # TODO: emulate it the same way as in Replica, that is sender must have 'node_name:inst_id' form
             replica_name = generateName(name, 0)
             handler = partial(self.network._send_message, replica_name)
+            write_manager = create_test_write_req_manager(name, genesis_txns)
             replica = ReplicaService(replica_name,
                                      validators,
                                      primary_name,
                                      self._timer,
                                      InternalBus(),
                                      self.network.create_peer(name, handler),
-                                     write_manager=create_test_write_req_manager(name, genesis_txns),
+                                     write_manager=write_manager,
                                      bls_bft_replica=MockBlsBftReplica())
+            # ToDo: For now, future_primary_handler is depended from the node.
```

Comment:
It looks like `FuturePrimariesBatchHandler` uses `node` in more places than just `_get_primaries`. In particular, it uses `node.requests`.